phel.html.markdown

```diff
@@ -0,0 +1,333 @@
+# => (+ (+ (+ 1 2) 3) 4)
+# => 10
+
+(reduce push [] '(3 2 1))
+# = (push (push (push [] 3) 2) 1)
+# => [3 2 1]
+
+# Functions
+#############
+
+# Use fn to create new functions. A function always returns its last statement.
+(fn [] "Hello World") # => :function
+
+# You need extra parens to call it
+((fn [] "Hello World")) # => "Hello World"
+
+# You can create a var using def
+(def x 1)
+x # => 1
+
+# Assign a function to a var
+(def hello-world (fn [] "Hello World"))
+(hello-world) # => "Hello World"
+
+# You can shorten this process by using defn
+(defn hello-world [] "Hello World")
+
+# The [] is the list of arguments for the function.
+(defn hello [name]
+  (str "Hello " name))
+(hello "Jens") # => "Hello Jens"
+
+# You can also use this shorthand to create functions:
+(def hello2 |(str "Hello " $1))
+(hello2 "Anna") # => "Hello Anna"
+
+# Functions can pack extra arguments up in a seq for you
+(defn count-args [& args]
+  (str "You passed " (count args) " args: " args))
+(count-args 1 2 3) # => "You passed 3 args: @[1 2 3]"
+
+# You can mix regular and packed arguments
+(defn hello-count [name & args]
+  (str "Hello " name ", you passed " (count args) " extra args"))
+(hello-count "Jesus" 1 2) # => "Hello Jesus, you passed 2 extra args"
+
+
+# Maps
+#############
+
+# Hash maps have faster lookups but don't retain key order.
+(type {:a 1 :b 2 :c 3})          # => :hash-map
+(type (hash-map :a 1 :b 2 :c 3)) # => :hash-map
+
+# Maps can use any hashable type as a key, but usually keywords are best
+# Keywords are like strings with some efficiency bonuses and they start with `:`
+(type :a) # => :keyword
+
+(def stringmap {"a" 1 "b" 2 "c" 3})
+stringmap  # => {"a" 1 "b" 2 "c" 3}
+
+(def keymap {:a 1 :b 2 :c 3})
+keymap # => {:a 1 :c 3 :b 2}
+
+# Retrieve a value from a map by calling it as a function
+(stringmap "a") # => 1
+(keymap :a)     # => 1
+
+# Keywords can be used to retrieve their value from a map, too!
+(:b keymap) # => 2
+
+# Don't try this with strings.
+# ("a" stringmap)
+# ...Exception: Call to undefined function a()
+
+# Retrieving a non-present key returns nil
+(stringmap "d") # => nil
+
+# Use put to add new keys to hash-maps
+(def newkeymap (put keymap :d 4))
+newkeymap # => {:a 1 :b 2 :c 3 :d 4}
+
+# But remember, phel types are immutable!
+keymap # => {:a 1 :b 2 :c 3}
+
+# Use unset to remove keys
+(unset keymap :a) # => {:b 2 :c 3}
+
+# Sets
+#############
+
+# A Set contains unique values in random order.
+
+(type (set 1 2 3)) # => :set
+(set 1 2 3 1 2 3 3 2 1 3 2 1) # => (set 1 2 3)
+
+# Add a member with push
+(push (set 1 2 3) 4) # => (set 1 2 3 4)
+
+# Remove one with unset
+(unset (set 1 2 3) 1) # => (set 2 3)
+
+# Test for existence by using the set as a function:
+((set 1 2 3) 1) # => 1
+((set 1 2 3) 4) # => nil
+
+# There are more functions like: count, union, intersection, difference, etc.
+
+
+# Useful forms
+#############
+
+# Logic constructs in clojure are just macros, and look like everything else
```

Comment:
Don't know how deep you want to go into this. But `if` is not a macro. It is a special form ;) But I think this is out of scope for this introduction.